patoolib/__init__.py

```diff
@@ -814,4 +814,31 @@ def get_inner_file_names_rar(file_path):
     'application/x-rar': get_inner_file_names_rar,
     'application/vnd.rar': get_inner_file_names_rar,
     'application/x-rar-compressed': get_inner_file_names_rar
-}
\ No newline at end of file
+}
+
+def get_inner_file_info(file_path):
+    files_list = []
+    if zipfile.is_zipfile(file_path):
+        return get_inner_file_info_zip(file_path, files_list)
+
+    if rarfile.is_rarfile(file_path):
+        return get_inner_file_info_rar(file_path, files_list)
+    # TODO: Add support to 7z archives using py7zr (python3)
```

Comment:
Once you add 7z support, we can use `get_inner_file_info` for the inner file names as well (and delete the functions `get_inner_file_names_zip`, `get_inner_file_names_rar`, `get_inner_file_names_7zip`).